etc/kayobe/ansible/scripts/nvmemon.sh

```diff
@@ -44,58 +66,69 @@ format_output() {
 nvme_version="$(nvme version | awk '$1 == "nvme" {print $3}')"
 echo "nvmecli{version=\"${nvme_version}\"} 1" | format_output
 
-# Get devices (DevicePath and PhysicalSize)
-device_info="$(nvme list -o json | jq -c '.Devices[] | {DevicePath: .DevicePath, PhysicalSize: .PhysicalSize}')"
+# Get devices (DevicePath, PhysicalSize and ModelNumber)
+device_info="$(nvme list -o json | jq -c '.Devices[] | {DevicePath, PhysicalSize, ModelNumber}')"
+
+# Convert device_info to an array
+device_info_array=()
+while IFS= read -r line; do
+  device_info_array+=("$line")
+done <<< "$device_info"
 
 # Loop through the NVMe devices
-echo "$device_info" | while read -r device_data; do
-  device=$(echo "$device_data" | jq -r '.DevicePath')
+for device_data in "${device_info_array[@]}"; do
+  device="$(echo "$device_data" | jq -r '.DevicePath')"
   json_check="$(nvme smart-log -o json "${device}")"
   disk="${device##*/}"
+  model_name="$(echo "$device_data" | jq -r '.ModelNumber')"
 
-  physical_size=$(echo "$device_data" | jq -r '.PhysicalSize')
-  echo "physical_size_bytes{device=\"${disk}\"} ${physical_size}"
+  physical_size="$(echo "$device_data" | jq -r '.PhysicalSize')"
+  echo "physical_size_bytes{device=\"${disk}\",model=\"${model_name}\"} ${physical_size}"
 
   # The temperature value in JSON is in Kelvin, we want Celsius
   value_temperature="$(echo "$json_check" | jq '.temperature - 273')"
-  echo "temperature_celsius{device=\"${disk}\"} ${value_temperature}"
+  echo "temperature_celsius{device=\"${disk}\",model=\"${model_name}\"} ${value_temperature}"
+
+  # Get the rated DWPD from the dictionary or default to 1 if not found
+  value_rated_dwpd="${rated_dwpd[$model_name]:-1}"
+  echo "rated_dwpd{device=\"${disk}\",model=\"${model_name}\"} ${value_rated_dwpd}"
 
   value_available_spare="$(echo "$json_check" | jq '.avail_spare / 100')"
-  echo "available_spare_ratio{device=\"${disk}\"} ${value_available_spare}"
+  echo "available_spare_ratio{device=\"${disk}\",model=\"${model_name}\"} ${value_available_spare}"
```

Comment:
Ideally we would have done the conversion of this to Python before this change. Jakub signed up for it ages ago but never started: https://stackhpc.atlassian.net/browse/INFRA-388

Reply:
Should we just roll with this in the meantime?

Reply:
That's fine